xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-api/src/main/java/org/xwiki/security/query/QueryConfiguration.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.security.query;
+
+import org.xwiki.component.annotation.Role;
+
+/**
+ * Configuration of queries made from Velocity templates.
+ *
+ * @version $Id$
+ * @since 13.7RC1
+ */
+@Role
+public interface QueryConfiguration
+{
+    /**
+     * @return the maximum size limit for a query to retrieve items.
```

Comment:
This is not clear to me. Do you mean: `the maximum number of query items to return`? It should also be explained in the javadoc what it's used for.

Reply:
This should go in a module config file, i.e. `SecurityConfiguration`.